### DISCUSSION TITLE
[Question][Hw4]Modifying given code

### DISCUSSION THREAD

**Discussion by jirheee** [prosyslab-classroom/cs348-information-security#569] [Q&A]
Are we allowed to modify given skeleton codes like fuzzer.ml or others?

**Comment by RiceBiscuits** [comment]
Basically, modifying existed skeleton codes except "Not implemented" is not allowed.

**Comment by AnHaechan** [comment]
@RiceBiscuits Then, is only changing the name of variables (a little bit) allowed?

**Comment by RiceBiscuits** [comment]
> @RiceBiscuits Then, is only changing the name of variables (a little bit) allowed?

Well.. I have no idea what can be changed just modifying the name of variables, but I will not recommend to you.

**Comment by seyoungsong** [comment]
Are we allowed to modify given skeleton codes related to **env**? #97 
To change env, we have to modify the env init code in `main.ml`.

Also, can we change the parameters of functions that we have to implement?

**Comment by RiceBiscuits** [comment]
> Are we allowed to modify given skeleton codes related to **env**? #97
> To change env, we have to modify the env init code in `main.ml`.
> 
> Also, can we change the parameters of functions that we have to implement?

It seems professor already allow to modify env. I think just adding init code is fine in that case.
I don't know what function do you want to modify, but please do not change the parameter of functions.

**Comment by ramel0915** [comment]
> Basically, modifying existed skeleton codes except "Not implemented" is not allowed.

I declared some functions in campaign.ml for mutate such as insert_random_string, delete_random_char. Is it okay..?

**Comment by Junseon00** [comment]
Yes, it's fine.

**Comment by ramel0915** [comment]
Thanks a lot!

**Comment by seyoungsong** [comment]
@RiceBiscuits 
Can we modify `mutate`? I need `seeds` to implement it.
If we can't, I can modify `env` to have seeds in it, or modify `run` to do some things that `mutate` should do.
But I think this is a bad solution...
